cache/elasticache_client.go

```diff
@@ -0,0 +1,180 @@
+package cache
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"time"
+
+	health "github.com/ONSdigital/dp-healthcheck/healthcheck"
+	"github.com/ONSdigital/dp-sessions-api/session"
+	"github.com/go-redis/redis"
+)
+
+var (
+	HealthyMessage       = "redis is OK"
+	ErrEmptySessionID    = errors.New("session id required but was empty")
+	ErrEmptySessionEmail = errors.New("session email required but was empty")
+	ErrEmptySession      = errors.New("session is empty")
+	ErrEmptyAddress      = errors.New("address is empty")
+	ErrEmptyPassword     = errors.New("password is empty")
+	ErrInvalidTTL        = errors.New("ttl should not be zero")
+)
+
+type ElasticacheClient struct {
+	client RedisClienter
+	ttl    time.Duration
+}
+
+// Config - config options for the redis client
+type Config struct {
+	Addr     string
+	Password string
```

Comment:
Being security conscientious can you add the `json:"-"` to the password field to ensure it's not accidentally logged out.